### DISCUSSION TITLE
maximum viewer possible ?

### DISCUSSION THREAD

**Discussion by tomtoom1978** [owncast/owncast#1684] [Q&A] [answered]
Today I used owncast with 325 simultanous viewers while +6 hours without any problem on a homemade intel celeron 3,5Ghz server, 1Gbps UP. I used passtrought video a 1024 kbps, video was good. And the CPU usage stay at 30%.
I will use this setup in the futur, and I would know and expect how much viewer this server can handle?!

I imagine around 900 viewers is possible?! Somebody test large viewers audience !?

![IMG_20220119_102932](https://user-images.githubusercontent.com/36678395/150199606-709a4306-b868-4eb0-9aca-53ed4b2e88f1.jpg)
![IMG_20220119_094405](https://user-images.githubusercontent.com/36678395/150199671-aae13106-948c-4f00-a773-2eca11960b60.jpg)

**Comment by gabek** [comment, marked as answer]
I unfortunately can't give you a simple answer, you're mostly going to be limited by your bandwidth for the number of people watching your video.  CPU usage won't really grow much with more viewers, though memory usage will grow a bit as more concurrent connections are being handled.  As far as chat, it's possible you might run into issues with chat when you're into the thousands of people using chat concurrently.

There's some notes here you might be interested in: https://owncast.online/docs/scaling/

**Comment by ogmkp** [comment]
Interresting.
@gabek 
I also have a transcoding machine that supports owncast at home, so to not to have a bottleneck and to hide my home IP address, I took a VM in the cloud with nginx reverse proxy + cache. On it I cached all TS, css, jpg, png, html and js.
I can't cache m3u8 because there is a cachebuster.
Do you have any best practices for nginx reverse proxy cache?

**Reply by gabek** [reply]
Owncast already sets caching headers, so as long as you respect those you should be in good shape.

**Reply by ogmkp** [reply]
> Owncast already sets caching headers, so as long as you respect those you should be in good shape.

To avoid overloading the internet connection, TS caching is convenient. Isn't there a risk of overloading owncast between the proxy and owncast with repeated m3u8 requests from the nginx IP?
